lib/iris/time.py

```diff
@@ -155,23 +155,21 @@ def __eq__(self, other):
                     other_attr = other.microsecond
                     if attr is not None and attr != other_attr:
                         result = False
+
             except AttributeError:
-                result = NotImplemented
+                result = other.__eq__(self)
+                if result is NotImplemented:
+                    # Equaliy is undefined between these objects.  We don't
+                    # want Python to fall back to the default `object`
+                    # behaviour (which compares using object IDs), so we raise
+                    # an exception here instead.
+                    fmt = "unable to compare PartialDateTime with {}"
+                    raise TypeError(fmt.format(type(other)))
```

Comment:
We previously used the `__cmp__` method to stop the equality comparison falling back to object identities.  But this method is [not a thing in python3](https://docs.python.org/3/whatsnew/3.0.html#ordering-comparisons).

My above solution works, but feels like a bit of a hack. Alternative solutions welcome!

Reply:
@rcomer Rings a bell... see https://github.com/Unidata/cftime/pull/81 :wink: 

This might explain this change in behaviour :thinking:

Reply:
@bjlittle yup the failures I saw were just for `==` and `!=` operators.  So I think `>=`, etc. were still raising the `TypeError` as there's no way to check that based on identities.

Reply:
@rcomer Okay, I'm fine with this.

It seems completely reasonable to me and maintains prior behaviour both within `iris` and for users :+1: